apps/explorer/src/app/App.tsx

```diff
@@ -14,67 +15,82 @@ import { growthbook, loadFeatures } from '../utils/growthbook';
 
 import styles from './App.module.css';
 
-const createQueryClient = () =>
-    new QueryClient({
-        defaultOptions: {
-            queries: {
-                refetchOnMount: false,
-                refetchOnWindowFocus: false,
-            },
+import { usePrevious } from '~/hooks/usePrevious';
+
+const queryClient = new QueryClient({
+    defaultOptions: {
+        queries: {
+            refetchOnMount: false,
+            refetchOnWindowFocus: false,
         },
-    });
+    },
+});
 
 // As a side-effect of this module loading, we start loading the features:
 loadFeatures();
 
 function App() {
     const [network, setNetwork] = useNetwork();
-    const [queryClient] = useState(createQueryClient);
+    const previousNetwork = usePrevious(network);
+
+    useEffect(() => {
+        if (network !== previousNetwork) {
+            queryClient.cancelQueries();
+            queryClient.clear();
+        }
+    }, [previousNetwork, network]);
 
-    // TODO: Verify this behavior:
     useEffect(() => {
-        queryClient.clear();
-    }, [network, queryClient]);
+        growthbook.setAttributes({
+            network,
+        });
+    }, [network]);
 
     return (
-        <GrowthBookProvider growthbook={growthbook}>
-            <QueryClientProvider client={queryClient}>
-                <NetworkContext.Provider value={[network, setNetwork]}>
-                    <div className={styles.app}>
-                        <Header />
-                        <main>
-                            <section className={styles.suicontainer}>
-                                <AppRoutes />
-                            </section>
-                        </main>
-                        <Footer />
-                    </div>
+        // NOTE: We set a top-level key here to force the entire react tree to be re-created when the network changes:
+        <React.Fragment key={network}>
```

Comment:
Is this necessary though? Does it fix any issues that happen at the moment? Anything that depends on `network` (or new data etc) ideally should render automatically, right?

Reply:
We discussed this in-person, but basically the reason why we want this is that changing networks really should invalidate everything, and effectively refresh the page. I debated making the actual page refresh but decided that this was probably enough.

Reply:
Thanks for clarifying!